### ISSUE TITLE
Compiler should warn on completely unnecessary cast (or IDE should gray it out to indicate that it does nothing).

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#489]
For example:

```c#
        public bool IsInInactiveRegion(SyntaxTree syntaxTree, int position, CancellationToken cancellationToken)
        {
            var csharpTree = syntaxTree as SyntaxTree;
            if (csharpTree == null)
            {
                return false;
            }

            return csharpTree.IsInInactiveRegion(position, cancellationToken);
        }
```

"syntaxTree as SyntaxTree" does nothing.  I believe if you casted to (SyntaxTree) then that would be marked appropriately by the IDE.  The 'as' check should be hte same.

**Comment by DustinCampbell** [comment]
This strikes me more as an analyzer scenario rather than something the compiler would do. The current Remove Unnecessary Cast diagnostic is an analyzer.

**Comment by srivatsn** [comment]
This would be adding a feature to remove unnecessary cast fix.

**Comment by jnm2** [comment]
Could this be marked `help wanted`?
